pkg/admiralctl/util/names/names.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2023 The KubeAdmiral Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package names
+
+import (
+	"fmt"
+	"hash"
+	"hash/fnv"
+	"strings"
+
+	"github.com/kr/pretty"
+	"k8s.io/apimachinery/pkg/util/rand"
+)
+
+// GeneratePolicyName generates the propagationPolicy name
+func GeneratePolicyName(namespace, name, gvk string) string {
```

Comment:
refer to `GenerateFederatedObjectName`, you can replace ftc name with resource group and kind
https://github.com/kubewharf/kubeadmiral/blob/e03546687e66454c4086d92788665f343b9dec1c/pkg/util/naming/naming.go#L61